integration_tests/suite/test_http_callback.py

```diff
@@ -896,3 +906,70 @@ def test_given_one_http_subscription_max_retries(self, subscription):
                 has_entries(status="failure"),
             ),
         )
+
+    @subscription(TEST_SUBSCRIPTION, tenant=USERS_TENANT)
+    def test_given_user_tenant_dont_trigger_on_other_tenant_event(self, subscription):
+        self.bus.publish(
+            trigger_event(),
+            headers={
+                'name': TRIGGER_EVENT_NAME,
+                'tenant_uuid': OTHER_TENANT,
+            },
+        )
+
+        assert_that(
+            calling(until.assert_).with_args(
+                self.make_third_party_verify_callback(
+                    request={'method': 'GET', 'path': '/test'}, count=1, exact=True
+                ),
+                timeout=10,
+                interval=0.5,
+            ),
+            raises(AssertionError),
+        )
```

Comment:
A timeout of 10 seconds is too long for a test asserting a non-event: we have to wait for the whole timeout for the test to pass.
One possible workaround is to send two events: one that should not trigger, and a second one that should trigger and arrive after the first, then assert that the second one triggered and not the first. This avoids having to wait for the whole timeout. This requires to be able to distinguish between the first and second event, of course.
If not possible, then maybe there is another way to test (unit-test?), or maybe we can remove the test altogether...